streams/src/main/java/org/apache/kafka/streams/errors/StreamsException.java

```diff
@@ -31,24 +37,34 @@ public class StreamsException extends KafkaException {
 
     private final static long serialVersionUID = 1L;
 
-    private TaskId taskId;
+    private final Set<TaskId> taskIds = new HashSet<>();
 
     public StreamsException(final String message) {
         this(message, (TaskId) null);
     }
 
     public StreamsException(final String message, final TaskId taskId) {
         super(message);
-        this.taskId = taskId;
+        taskIds.add(taskId);
+    }
+
+    public StreamsException(final String message, final Set<TaskId> taskIds) {
```

Comment:
Should this be `protected` or package-private instead of `public` ?

Reply:
Switched to package-private.